src/storage-client/src/source/persist_source.rs

```diff
@@ -134,38 +144,39 @@ where
         Arc::new(metadata.relation_desc),
         Arc::new(UnitSchema),
     );
-    let rows = decode_and_mfp(&fetched, &name, until, map_filter_project, yield_fn);
+    let rows = decode(&fetched, &name, until.clone(), yield_fn).as_collection();
+
+    let rows = match consolidation {
+        Consolidation::Maybe => rows,
+        Consolidation::Consolidated => rows.consolidate(),
```

Comment:
This uses the default arrangement, which is backed by heap-allocated rows. Instead, it should use the `RowSpine` arrangement spine to make sure it uses columnated rows.